polygeist/tools/cgeist/driver.cc

```diff
@@ -824,6 +840,10 @@ static LogicalResult compileModule(mlir::OwningOpRef<mlir::ModuleOp> &Module,
                  << "*** Dumped MLIR in file '" << Output << "' ***\n");
     }
   } else {
+    // Host code is never output for non-MLIR format. If this option is set, the
+    // host code should have already been removed.
+    if (!SYCLDeviceOnly)
```

Comment:
Also need to check that the current module is for SYCL? 
There are `check-cgeist` failures, I guess due to erasing host code for non SYCL test cases.

Reply:
Yes, that was the error indeed... Thanks!